src/fdc/MSXtar.cc

```diff
@@ -104,6 +104,10 @@ void MSXtar::parseBootSector(const MSXBootSector& boot)
 	chrootSector = rootDirStart;
 	dataStart = rootDirStart + nbRootDirSectors;
 
+	if (dataStart + sectorsPerCluster > boot.nrSectors) {
+		throw MSXException("Cluster count too small: ", (int(boot.nrSectors) - dataStart) / sectorsPerCluster);
+	}
+
 	clusterCount = (boot.nrSectors - dataStart) / sectorsPerCluster;
 	clusterCount = std::min(clusterCount, FAT12_MAX_CLUSTER_COUNT);
```

Comment:
Suggestion for a simplification: move the check down two lines and write it as `if (clusterCount <= 0)`.
Ah, but `clusterCount` is an unsigned variable so that won't work. Is that the reason why you wrote it like this?

The error message itself may not be the best. It will either print 0 or a very large value. Maybe formulate it as "Too few sectors: n"?

Reply:
Yes that’s the reason.

When you say a very large value, I probably got my signed / unsigned implicit conversion rules wrong? Since the intent was that the initial `int()` would make the rest of the expression operate on signed numbers.

Anyway I’ll reformulate it, yes.

Reply:
Ah, I overlooked that `int()` conversion. For some reason I thought the calculation was identical to the one on line 111.

But `(<int> <binary-operation> <unsigned>)` results in an `unsigned` (and I'm assuming `dataStart` and/or `sectorsPerCluster` is `unsigned`?).

Reply:
Yeah it’s probably wrong, I should’ve looked up the implicit conversion rules. I rushed it at the last minute before going to bed. Anyway, changed now to your suggestion to just call out the sector count as being too few.